app/models/mentor/application.rb

```diff
@@ -0,0 +1,118 @@
+module Mentor
+  class Application
+    include ActiveModel::Model
+
+    DB = ActiveRecord::Base.connection
+
+    attr_accessor :id, :team_name
+    attr_accessor :project_id, :project_name
+    attr_accessor :project_plan, :why_selected_project
+    attr_accessor :student0, :student1
+    attr_accessor :first_choice
+
+    def choice
+      first_choice ? 1 : 2
+    end
+
+    def student0=(attrs)
+      attrs     = JSON.parse(attrs).instance_eval(&studentize)
```

Comment:
Holy :cow:  – very elegant!

It took me some time to wrap my mind around it, though, what with the 3 levels of indirection. Do you see a way to rewrite this in a less inside-out object modification manner?

Does this work:

```ruby
def student1=(attrs)
  attrs = JSON.parse(studentize(attrs))
end
# …
def studentize(attrs)
  attrs.keys.each{ |k| attrs[k.gsub(/student(0|1)_application_/, '')] = attrs.delete(k) } }
end
```

But that still modifies the passed `attrs` value, no? It would rather work with `#dup`s. Or is the mutation a desired/required effect?